### DISCUSSION TITLE
Critical CSS Extraction

### DISCUSSION THREAD

**Discussion by lilyntalmers** [gatsbyjs/gatsby#36633] [Help]
Google's lighthouse audit gives us direction to inline critical CSS to improve performance.

On Gatsby's homepage it says "It's in our DNA to be fast" but I couldn't find a single plugin for inlining critical CSS. I also couldn't find any documentation about any inbuilt feature for this.

[Critters](https://github.com/GoogleChromeLabs/critters) (created by Google) solves this exact problem.

@KyleAMathews seemed really [interested about Critters](https://twitter.com/_developit/status/1001935620854476802) but couldn't find any documentation or plugin related to it.

Even Astro seems to have a [plugin](https://www.npmjs.com/package/astro-critters) for this.

Am I missing something?

**Comment by cometkim** [comment]
As far as I know, the default policy of the Gatsby plugins is "all chunks" injection. Subsequent optimizations are left to the plugin or the user to decide.

[gatsby-plugin-linaria](https://github.com/cometkim/gatsby-plugin-linaria#critical-css-extraction) supports critical css extraction as an opt-in feature.

As I mentioned in the docs, determining when to load the rest of the styles in Gatsby is rather difficult when you use zero-runtime library.

**Reply by lilyntalmers** [reply]
Thank you @cometkim for your reply. [gatsby-plugin-linaria](https://github.com/cometkim/gatsby-plugin-linaria#critical-css-extraction) feels to me the best solution for Gatsby till now.

I already took a look but I'm having hard time understanding how it'll choose which CSS classes needs to be extracted as `critical css`. I mean, do I need to mark certain part of the CSS to be extracted as `critical css` or will the plugin automatically choose by itself if I opt-in for the feature?

**Reply by cometkim** [reply]
It is automatically collected on the server when it rendered.

See https://github.com/callstack/linaria/blob/master/docs/API.md#server-apis-linariaserver
